File: free_tasks/ft_js/palindrome_number.js
class Solution {
    static isPalindrome(x) {
        let useX = x;
        if (x <= 0) {
            return false;
        }
        let reversedNum = 0;
        while (useX !== 0) {
            const digit = useX % 10;
            reversedNum = reversedNum * 10 + digit;
            useX = Math.floor(useX / 10);
        }
        return x === reversedNum;
    }
}

const x = 1221;
console.log(Solution.isPalindrome(x));

File: free_tasks/ft_py/palindrome_number.py
class Solution:
    x = 1221
    def isPalindrome(x: int) -> bool:
        use_x = x 
        if x <= 0:
            return False
        reversed_num = 0
        while use_x != 0:
            digit = use_x % 10
            reversed_num = reversed_num * 10 + digit
            use_x //= 10
        if x == reversed_num:
            return True
    print(isPalindrome(x))    